src/transformers/models/gptj/modeling_gptj.py

```diff
@@ -113,9 +98,8 @@ def __init__(self, config):
         self.v_proj = nn.Linear(self.embed_dim, self.embed_dim, bias=False)
         self.q_proj = nn.Linear(self.embed_dim, self.embed_dim, bias=False)
         self.out_proj = nn.Linear(self.embed_dim, self.embed_dim, bias=False)
-        self.rotary_dim = None
-        if config.rotary_dim is not None:
-            self.rotary_dim = config.rotary_dim
+        self.rotary_dim = config.rotary_dim
+        self.embed_positions = lookup_embed_positions
```

Comment:
I've memoized the precomputed positions at the model level to avoid generating/storing a redundant copy for every layer. Perhaps it's small enough (a few MB I think) that it's not a concern? If so I can revert this.

I think we also want these to be on the target device which I wasn't sure is necessarily known at construction time, hence the lookup taking a device from generation-time tensors.

Reply:
Agreed that is more efficient! However, in `transformers`, we prioritize consistent APIs and classes over efficiency, and GPT-J's classes would become an outlier with that extra init parameter :D